Fuzzers: `AK/Coroutine.h` causes the fuzzer build to fail

The fuzzer build is currently failing with the following error:
```
/src/serenity/Meta/Lagom/../../AK/Coroutine.h:11:10: fatal error: 'coroutine' file not found
   11 | #include <coroutine>
      |          ^~~~~~~~~~~
1 error generated
```
Here is the full [build log](https://oss-fuzz-build-logs.storage.googleapis.com/log-9502fe49-7a2f-4d5c-bdcd-408610a24a8c.txt).

--- a/Meta/Lagom/BuildFuzzers.sh
+++ b/Meta/Lagom/BuildFuzzers.sh
@@ -47,7 +47,10 @@ unset CXXFLAGS
 export AFL_NOOPT=1
 
 if [ "$#" -gt "0" ] && [ "--oss-fuzz" = "$1" ] ; then
-    CXXFLAGS="$CXXFLAGS -DOSS_FUZZ=ON"
+    # The oss-fuzz runner image (as of June 2024) has libstdc++ 9, which lacks the
+    # <coroutine> header. Use the more modern libc++ shipped with oss-fuzz's Clang.
+    # It's already set in CXXFLAGS_SAVE.
+    CXXFLAGS="$CXXFLAGS -DOSS_FUZZ=ON -stdlib=libc++"
 fi
 
 # FIXME: Replace these CMake invocations with a CMake superbuild?

